### PR TITLE
Add Text class

### PR DESCRIPTION
**Pull request by bensuperpc**
Add Text class

Signed-off-by: Bensuperpc <bensuperpc@gmail.com>

### CODE REVIEW

File: include/Text.hpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+     * @param text Text to initialize.
+     * @param posX X position of the text.
+     * @param posY Y position of the text.
+     * @return void.
+     */
+    Text(const std::string& text, const float posX, const float posY)
+    {
+        _text = text;
+        _position.x = posX;
+        _position.y = posY;
```
Comment (RobLoach) [inline include/Text.hpp:37]: Initializer lists for the constructors may be a good call here https://en.cppreference.com/w/cpp/language/constructor#Initialization_order

**Review by RobLoach — COMMENTED** [review]
This is looking great! Love the idea of having some Text-specific functions in here. Could likely build upon it afterwards for some of the text manip functions too. Thanks so much!

File: include/Text.hpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        ::DrawTextPro(font, text.c_str(), position, origin, rotation, fontSize, spacing, color);
+    }
+
+  private:
+    std::string _text = "";
+    float _fontSize = 10;
+    Font _font = ::GetFontDefault();
+    Color _color = WHITE;
+    Vector2 _position = {0.0, 0.0};
+    float _spacing = 0;
```
[the remark this answers is not in the document]
Comment (bensuperpc) [reply include/Text.hpp:153]: > I like having font, text, size, color, and spacing in here, as those are properties of the text... Unconvinced on the `_position` though. The `Texture` class, for instance, doesn't have position in there.

I would like to do as with other libraries (SFML, ...), all properties are in the class, and all you have to do is call the function for the display, i can add other draw functions to keep the several possibilities.
I was thinking of doing this with the other classes as well ^^

[the remark this answers is not in the document]
Comment (RobLoach) [reply include/Text.hpp:161]: While that would be really cool, my hope to keep the classes data-compatible with the raylib structs. You could, however, create a new class that inherits from them with a Vector2 position on each one :wink: ... Could be useful in a raylib-cpp-extras or something!

**Comment by RobLoach** [comment]
Updated the PR over at https://github.com/bensuperpc/raylib-cpp/pull/1

**Comment by bensuperpc** [comment]
There are new warnings that have appeared: https://github.com/RobLoach/raylib-cpp/runs/6449081713?check_suite_focus=true#step:5:128

Doesn't seem to matter in this case, in other cases, it can be a problem (ex if Text::font depends on Text::color), after that makes the other warnings less readable :/

File: include/Text.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bensuperpc) [reply include/Text.hpp]: > While that would be really cool, my hope to keep the classes data-compatible with the raylib structs. You could, however, create a new class that inherits from them with a Vector2 position on each one wink ... Could be useful in a raylib-cpp-extras or something!

I plan to complete this library, make another one that will complete it for my future games ^^

**Pull request merged**